### DISCUSSION TITLE
way to export an aaSequence even if it's not part of the gene features used to group clonotypes ?

### DISCUSSION THREAD

**Discussion by Just08** [milaboratory/mixcr#1141] [Q&A] [answered]
Hi,

I managed to produce a table of clonotypes grouped by VDJRegion and to obtain the amino acid sequences of all the genes features characterized by VDJRegion in the table output by exportClones.
Now I'm grouping the clonotypes by CDR3 on the same aligned sequences, but I don't know how to get the amino acid sequences of all the genes features characterized by VDJRegion even though it's  not part of the gene features used to group clonotypes, is there a way?

Thanks in advance.

**Comment by mizraelson** [comment, marked as answer]
Can you please share the exact commands that you used and mixcr version?

If I understand correctly, you have used two different assembling features: VDJRegion and CDR3. The latter does not include features upstream and downstream of CDR3. Its impossible to include them because some nucleotides might be ambiguous.
Lets say you have two clones with identical CDR3 sequence: one with a hypermutation in FR2 region, the other one with a naive sequence. When assembling by VDJRegion you will get two separate clones, but if you assemble sequences by CDR3 region, reads from both clones will be aggregated together, and we wouldn't know what nucleotide to export for FR2 region (naive or the one with mutation as we have reads for both).

**Comment by Just08** [comment]
Thanks for your answer.

Since yesterday I used :

For clonotype grouping by CDR3 sequence + same V and J assignments : 
     
```
     mixcr assemble \\
    --write-alignments \\
    --dont-infer-threshold \\
    -Passembler.minRecordsPerConsensus=5 \\
    -OseparateByV=true \\
    -OseparateByJ=true \\
    --json-report CDR3/!{refined_vdjca.SimpleName}/!{refined_vdjca.SimpleName}.json \\
    --report CDR3/!{refined_vdjca.SimpleName}/!{refined_vdjca.SimpleName}.txt \\
    !{refined_vdjca} \\
    CDR3/!{refined_vdjca.SimpleName}/!{refined_vdjca.SimpleName}.clna

    mixcr assembleContigs \\
    -OassemblingRegions=VDJRegion \\
    -OpostFiltering.type=OnlyFullyAssembled \\
    CDR3/!{refined_vdjca.SimpleName}/!{refined_vdjca.SimpleName}.clna \\
    CDR3/!{refined_vdjca.SimpleName}/!{refined_vdjca.SimpleName}.clns
```

The assembleContigs step fills other aasequences columns except for non-complete sequences. Is it right to do it this way ?

**Comment by mizraelson** [comment]
Hi, `assembleContigs` is used for fragmented sequences assembly ( when lots of sequences don't cover the initial assembling feature (usually its CDR3). In your case you will actually end up splitting the clones initially assembled by CDR3 if there are some ambiguous variants in other regions, so the data will be kinda assembled by `VDJRegion`. If you work with target amplicon TCR/BCR repertoire data - running these commands should produce more or less the same result as using `VDJRegion` assembly feature on `mixcr assemble` step as you did earlier.
